src/pages/docs/error-handling.mdx

```diff
@@ -3,156 +3,207 @@ title: Error Handling
 
-<Section>
-<Left>
-Additional Error Properties
-----------------------------------------
-Before throwing an error, you can set additional properties that provide more context. 
-
-Doing this enables our platform to respond appropriately and provide meaningful feedback to users.
-
 <Reference>
-  <Field name="code" type="string">
+  <Field name="message" type="string">
     <Description>
-      A string that identifies the type of error. Most errors that occur in Node.js already have a code, but if you're throwing a new error, then we recommend that you use one of our pre-defined [error codes](#error-codes) below.
+      This is the message that will be displayed to the end users of the
+      platform.
     </Description>
   </Field>
-
-  <Field name="statusCode" type="number">
+  <Field name="details">
+    <Type>[ErrorDetail](#error-detail) | [ErrorDetail](#error-detail)[]</Type>
     <Description>
-      The numeric status code associated with the error, if any. For errors that originate from an HTTP response, this should be the HTTP status code (e.g. 400, 404, 500).
+      This contains additional information that needs to be logged along side
+      the error.
     </Description>
   </Field>
+</Reference>
+
+## Unauthorized Error
 
-  <Field name="externalErrors" type="string[]">
+This error lets a user know that they need to update their credentials in order to continue using their connection.
+
+```javascript highlight=5
+import { UnauthorizedError } from "@shipengine/connect-runtime";
+// ...
+const response = await axios.post(THIRD_PARTY_API_URL, { credentials });
+if (response.statusCode === 401) {
+  throw new UnauthorizedError("Invalid Credentials");
+}
+```
+
+<Reference>
+  <Field name="message" type="string">
     <Description>
-      If the error is the result of calling an external service, then put any error messages that were returned by the service here.
+      This is the message that will be displayed to the end users of the
+      platform.
     </Description>
   </Field>
-
-  <Field name="externalWarnings" type="string[]">
+  <Field name="details">
+    <Type>[ErrorDetail](#error-detail) | [ErrorDetail](#error-detail)[]</Type>
     <Description>
-      If the error is the result of calling an external service, then put any warning messages that were returned by the service here.
+      This contains additional information that needs to be logged along side
+      the error.
     </Description>
   </Field>
+</Reference>
+## External Server Error
 
-  <Field name="originalError">
-    <Type>
-      [Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error)
-    </Type>
+This type of error lets the user know that there is an issue with the outside world that we cannot resolve.
+
+```javascript highlight=5
+import { ExternalServerError } from "@shipengine/connect-runtime";
+// ...
+const response = await axios.post(THIRD_PARTY_API_URL, { credentials });
+if (response.statusCode > 500) {
```

Comment:
nit, but unless you confgure your axios a certain way it will throw an error instead of return a response with status code > 300. I think removing to call to axios is fine and just have response as an object. People will pick up on it.